src/zenml/models/v2/core/model.py

```diff
@@ -0,0 +1,314 @@
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Models representing models."""
+
+from datetime import datetime
+from typing import TYPE_CHECKING, ClassVar, List, Optional, Union
+from uuid import UUID
+
+from pydantic import BaseModel, Field
+
+from zenml.constants import STR_FIELD_MAX_LENGTH, TEXT_FIELD_MAX_LENGTH
+from zenml.models.v2.base.scoped import (
+    WorkspaceScopedFilter,
+    WorkspaceScopedRequest,
+    WorkspaceScopedResponse,
+    WorkspaceScopedResponseBody,
+    WorkspaceScopedResponseMetadata,
+)
+
+if TYPE_CHECKING:
+    from zenml.model.model_version import ModelVersion
+    from zenml.models.tag_models import TagResponseModel
+
+# ------------------ Request Model ------------------
+
+
+class ModelRequest(WorkspaceScopedRequest):
+    """Request model for models."""
+
+    name: str = Field(
+        title="The name of the model",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    license: Optional[str] = Field(
+        title="The license model created under",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    description: Optional[str] = Field(
+        title="The description of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    audience: Optional[str] = Field(
+        title="The target audience of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    use_cases: Optional[str] = Field(
+        title="The use cases of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    limitations: Optional[str] = Field(
+        title="The know limitations of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    trade_offs: Optional[str] = Field(
+        title="The trade offs of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    ethics: Optional[str] = Field(
+        title="The ethical implications of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    tags: Optional[List[str]] = Field(
+        title="Tags associated with the model",
+    )
+
+
+# ------------------ Update Model ------------------
+
+
+class ModelUpdate(BaseModel):
+    """Model update model."""
+
+    license: Optional[str]
+    description: Optional[str]
+    audience: Optional[str]
+    use_cases: Optional[str]
+    limitations: Optional[str]
+    trade_offs: Optional[str]
+    ethics: Optional[str]
+    add_tags: Optional[List[str]]
+    remove_tags: Optional[List[str]]
+
+
+# ------------------ Response Model ------------------
+
+
+class ModelResponseBody(WorkspaceScopedResponseBody):
+    """Response body for models."""
+
+    tags: List["TagResponseModel"] = Field(
```

Comment:
should tags be in metadata or body? Probably @avishniakov or @htahir1 know this best?

For artifacts I added them to metadata, we should probably keep that consistent

Reply:
Wouldnt it make sense to have them outside? I lean towards that

Reply:
How I understand this whole hydration concept: what is in the body (hydrate=False) belongs to listers, and what is in metadata (hydrate=True) belongs to getters. So Tags belong to the body, cause it is used in lists by the frontend.